### DISCUSSION TITLE
Iobroker Mqtt Verbindung zu Solectrus Mqtt nicht möglich

### DISCUSSION THREAD

**Discussion by Ritschy2000** [solectrus/solectrus#3686] [Q&A] [answered]
Hallo, 

ich versuche schon seit mehreren Tagen Solectrus korrekt zum Laufen zu bekommen, scheitere aber leider immer an der Mqtt Verbindung 
zwischen iobroker und Solectrus.

folgender Ablauf:
- Linux (Debian 12) in Proxmox installieren (als VM und LXC getestet)
- Docker im Linux installieren 
- über den Solectrus Konfigurator alles hinterlegen und nach erstellter Read.me weiter installieren,
  es werden keine Fehlermeldungen oder ähnliches angezeigt, Docker-Container werden alle angelegt (insgesamt 7),
  starten alle und zeigen keine Fehler
- im Iobroker Mqtt Client installieren (IP von der VM bzw. LCX; Port 1889, da andere belegt) und Werte wie im Konfigurator
   hinterlegen

**Ich kann auf das Solcetrus Dashboard normal zugreifen, bekomme aber keine Daten.
im Iobroker gibt der Mqtt Client die Fehlermeldung aus, dass er den Client (also Solectrus) nicht erreichen kann.**

Habe bereits mit einem anderen Mqtt Broker getestet, da kommen die Daten aus dem Iobroker an. 
( sollte also im Iobroker alles korrekt eingestellt sein)

Wenn ich im Linux in den laut Anleitung angelegten Solectrus Ordner schaue, sind auch die compose und env Datei vorhanden und
auch die Daten korrekt angelegt. 
Ein `docker compose logs mqtt-collector` im Solectrus Ordner bring folgende Fehlermeldung:
`mqtt-collector-1  | 2024-11-16 21:00:10 +0100: Connection refused - connect(2) for "192.168.10.10" port 1889, will retry again in 5 seconds...
`  (192.168.10.10 ist die IP vom Iobroker mit dem entsprechenden Port)

Kenne mich mit Docker leider überhaupt nicht aus (über Portainer gibt es zumindest keine Fehlermeldung).
Muss eventuell noch Port 1889 freigegeben werden, da es nicht der Standard Port ist?? Wenn ja, wie??
Habe auch schon die interne IP aus dem Dockercontainer des Mqtt Collectors im Iobroker eingegeben (172.18.0.2), leider ohne Erfolg.

Hat jemand eine Idee, was ich noch probieren könnte???

Vielen Dank und viele Grüße

**Comment by ledermann** [comment, marked as answer]
Dank deiner sehr ausführlichen Beschreibung 👍 lässt sich klar erkennen, wo das Problem ist: Es wird bei ioBroker ein **MQTT-Server/Broker** benötigt, kein **Client/subsriber**. Der MQTT-Collector von SOLECTRUS will die Messwerte bei einem Server abholen, nicht dorthin senden. Der Port, auf dem ioBroker üblicherweise reagiert, ist `1883`. Dieser Port (oder ein selbst gewählter) muss vom MQTT-Collector aus erreichbar sein.

Damit Messwerte in SOLECTRUS sichtbar werden, müssen die Messwerte deiner Geräte (Wechselrichter, Wallbox etc.) irgendwie in ioBroker landen. Das erledigen meist die dortigen Adapter. Wenn das funktioniert, dann sorgt der MQTT-Broker des ioBroker dafür, dass fremde Clients (wie der MQTT-Collector von SOLECTRUS) sie dort abholen können.

Du solltest schrittweise vorgehen:

1) Sicherstellen, dass Messwerte vom ioBroker bei deinen Geräten abgeholt werden (Adapter installieren, Instanzen konfigurieren). Im ioBroker unter "Objekte" müssen dann jederzeit aktuelle Werte sichtbar sein.
2) MQTT-Broker einrichten (als Server, nicht als Client) und dafür sorgen, dass Topics bereitgestellt werden.
3) Den MQTT-Collector von SOLECTRUS so einstellen, dass er sich mit dem MQTT-Broker verbindet (IP-Adresse, Port, ggfs. Username/Passwort) und die passenden Topics abonniert. Im Docker-Log des MQTT-Collectors dürfen dann nur Erfolgsmeldungen auftauchen.

**Reply by Ritschy2000** [reply]
Vielen Dank für die ausführliche und schnelle Antwort.
Kaum macht man es richtig, schon funktioniert es. :)
Jetzt macht es für mich auch Sinn, warum evcc nicht direkt verbunden werden kann.

**Eine kurze andere Frage hätte ich noch:** 
Wärmepumpen mit separatem Stromzähler = Wärmepumpen-Tarif können aktuell noch nicht eingebunden werden, oder?

**Reply by ledermann** [reply]
Super, freut mich, dass es geklappt.

Unterstützung für Wärmepumpentarife gibt es bislang nicht. Ich habe #3692 angelegt.
